src/markdown-pages/collect-data/opentelemetry-manual/instrument.mdx

```diff
@@ -0,0 +1,641 @@
+        return True
+
+    return False
+```
+
+</>
+
+<br />
+
+A tracer provider is used for holding configurations and for building [tracers](https://opentelemetry.io/docs/reference/specification/trace/api/#tracer). Tracers are then used for creating [spans](https://opentelemetry.io/docs/reference/specification/trace/api/#span). Spans collect information about an operation or process.
+
+Here, you created a tracer provider with a resource. A [resource](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md) describes a service as a collection of attributes. In this resource, you specified name of your service as "speedeedeebee".
+
+Finally, configure your API to use your new tracer provider.
+
+</Step>
+
+<Step>
+
+Add a [span processor](https://opentelemetry.io/docs/reference/specification/trace/sdk/#span-processor):
+
+<>
+
+```python fileName=db.py
+import logging
+from grpc import Compression
+from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+provider = TracerProvider(
+    resource=Resource.create({"service.name": "speedeedeebee"})
+)
+provider.add_span_processor(
+    BatchSpanProcessor(
+        OTLPSpanExporter(compression=Compression.Gzip)
+    )
+)
+trace.set_tracer_provider(provider)
+
+class DuplicateKeyError(Exception):
+    pass
+
+class KeyDoesNotExistError(Exception):
+    pass
+
+db = {}
+
+def read(key):
+    """Read key from the database."""
+    global db
+
+    try:
+        value = db[key]
+        logging.debug("Successful read")
+        return value
+    except KeyError as ke:
+        msg = f"Key `{key}` doesn't exist"
+        logging.debug(msg)
+        raise KeyDoesNotExistError(msg)
+
+def create(key, value):
+    """Write key:value to the database."""
+    global db
+
+    if key in db:
+        msg = f"Key `{key}` already exists"
+        logging.debug(msg)
+        raise DuplicateKeyError(msg)
+
+    db[key] = value
+    logging.debug("Successful create")
+    return value
+
+def update(key, value):
+    """Update key in the database."""
+    global db
+
+    if key in db:
+        db[key] = value
+        logging.debug("Successful update")
+        return value
+
+    msg = f"Key `{key}` doesn't exist"
+    logging.debug(msg)
+    raise KeyDoesNotExistError(msg)
+
+def delete(key):
+    """Delete key from the database."""
+    global db
+
+    if key in db:
+        del db[key]
+        logging.debug("Successful delete")
+        return True
+
+    return False
+```
+
+</>
+
+<br />
+
+A span processor processes span data before exporting it to a telemetry consumer. The `BatchSpanProcessor` you used here batches spans before exporting them. This reduces the number of requests you send to New Relic.
```

Comment:
Same applies here. What is a span should be explained before adding one. I feel more comfortable reading a piece that follows what, why, how structure.